platform/solr/solr-factory/src/test/groovy/org/codice/solr/factory/impl/SolrCloudClientFactorySpec.groovy

```diff
@@ -0,0 +1,558 @@
+      0 * cloudClient.uploadConfig(*_)
+
+    then: "verify zookeeper is consulted to see if the collection exists"
+      1 * cloudClient.request({ it instanceof CollectionAdminRequest.List }, null) >> listResponse
+      1 * listResponse.get('collections') >> Collections.emptyList()
+
+    and: "the collection is created"
+      1 * cloudClient.request({
+        ((it instanceof CollectionAdminRequest.Create)
+            && (it.numShards == SOLR_SHARD_COUNT)
+            && (it.maxShardsPerNode == SOLR_MAX_SHARDS_PER_NODE)
+            && (it.replicationFactor == SOLR_REPLICATION_FACTOR)
+            && (it.collectionName == CORE))
+      }, null) >> createResponse
+      1 * createResponse.get('success') >> true
+
+    and: "verify zookeeper is consulted to see if the collection was created"
+      1 * clusterState.hasCollection(CORE) >> true
+
+    and: "verify zookeeper is consulted to see if the shards were started"
+      1 * clusterState.getSlices(CORE) >> ['shard'] * SOLR_SHARD_COUNT
+
+    and: "close() is never called on the underlying client"
+      0 * cloudClient.close()
+  }
+
+  @Unroll
+  def 'test creating a Solr cloud client when configuration is uploaded and the collection already exists and system property solr.data.dir is #data_dir_is'() {
+    given:
+      System.setPropertyIfNotNull("solr.data.dir", solr_data_dir)
+
+    and:
+      def zkClient = Mock(SolrZkClient)
+      def listResponse = Mock(NamedList)
+      def cloudClient = Mock(CloudSolrClient) {
+        getZkStateReader() >> Mock(ZkStateReader) {
+          getZkClient() >> zkClient
+        }
+      }
+      def factory = Spy(SolrCloudClientFactory)
+
+    when:
+      def createdClient = factory.createSolrCloudClient(SOLR_CLOUD_ZOOKEEPERS, CORE)
+
+    then: "verify the Solr cloud client is created"
+      1 * factory.newCloudSolrClient(SOLR_CLOUD_ZOOKEEPERS) >> cloudClient
+
+    and: "the returned client is the one we created"
+      createdClient.is(cloudClient)
+
+    and: "it is being connected"
+      1 * cloudClient.connect() >> null
+
+    then: "verify zookeeper is consulted to see if the configuration exists"
+      1 * zkClient.exists("/configs/$CORE", true) >> false
+
+    and: "the config is uploaded to zookeeper"
+      1 * cloudClient.uploadConfig(*_) >> null
+
+    then: "verify zookeeper is consulted to see if the collection exists"
+      1 * cloudClient.request({ it instanceof CollectionAdminRequest.List }, null) >> listResponse
+      1 * listResponse.get('collections') >> Collections.singletonList(CORE)
+
+    and: "the collection is never created"
+      0 * cloudClient.request({ it instanceof CollectionAdminRequest.Create }, null)
+
+    and: "close() is never called on the underlying client"
+      0 * cloudClient.close()
+
+    and: "the config store is initialized and its data directory was or wasn't updated"
+      ConfigurationStore.instance.dataDirectoryPath == data_dir
```

Comment:
❓Couldn't we use `solr_data_dir` here instead of using a different `data_dir` variable?